ngraph/core/src/descriptor/tensor.cpp

```diff
@@ -46,14 +43,8 @@ void descriptor::Tensor::set_element_type(const element::Type& element_type)
 void descriptor::Tensor::set_partial_shape(const PartialShape& partial_shape)
 {
     m_partial_shape = partial_shape;
-    if (m_partial_shape.is_static())
-    {
-        m_shape = m_partial_shape.to_shape();
-    }
-    else
-    {
-        m_shape = Shape{};
-    }
+    std::lock_guard<std::mutex> guard(shape_mutex);
```

Comment:
this lock is also not needed as we're not targetting to have set_partial_shape thread-safe with get_shape